client/src/components/DatedItem/index.js

```diff
@@ -0,0 +1,29 @@
+import React, { Component } from 'react';
+
+class DatedItem extends Component {
+
+  constructor(props) {
+    super(props);
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  //invoke
+  clickHandler() {
```

Comment:
I am not 100% sure but I think that if you make this function as an arrow function and assign it to a const 'clickHandler', then you wouldn't even need to write the constructor for this class.
This is because the click handler function is the only thing inside the constructor and with arrow function you would bind the context of click handler to the DatedItem component automatically.